src/os/bluestore/BlueFS.cc

```diff
@@ -1775,15 +1777,13 @@ int BlueFS::_allocate(uint8_t id, uint64_t len,
     return r;
   }
 
-  uint64_t hint = 0;
-  if (!ev->empty()) {
-    hint = ev->back().end();
-  }
+  static std::atomic<std::uint64_t> hint { 0 };
```

Comment:
Wondering if we can have a hint as an allocator's member instead? Do different calls on the same allocator really use different hints? IMO external hints make sense only when the answer is 'yes'... And even if that's true - 'default' (i.e. one that is preserved as static/thread_local) hint can be encapsulated and user supplied one can be made optional...